cpp/src/arrow/compute/kernels/scalar_string_ascii.cc

```diff
@@ -2063,7 +2063,12 @@ struct RegexSubstringReplacer {
         regex_find_("(" + options_.pattern + ")", MakeRE2Options<Type>()),
         regex_replacement_(options_.pattern, MakeRE2Options<Type>()) {}
 
-  Status ReplaceString(re2::StringPiece s, TypedBufferBuilder<uint8_t>* builder) const {
+  Status ReplaceString(std::string_view s, TypedBufferBuilder<uint8_t>* builder) const {
+    re2::StringPiece piece(s.data(), s.length());
+    return ReplaceStringImpl(piece, builder);
+  }
+
+  Status ReplaceStringImpl(re2::StringPiece s, TypedBufferBuilder<uint8_t>* builder) const {
```

Comment:
I think this should fix the lint error:
```suggestion
  Status ReplaceStringImpl(re2::StringPiece s,
                           TypedBufferBuilder<uint8_t>* builder) const {
```

Reply:
Fixed and pushed. I lost my pre-commit hook and missed this.